src/pages/media/AttachmentModalScreen/AttachmentModalBaseContent.tsx

```diff
@@ -0,0 +1,584 @@
+    fallbackRoute?: Route;
+
+    /** Determines if the attachment is invalid or not */
+    isAttachmentInvalid?: boolean;
+
+    /** Determines if the attachment is invalid or not */
+    attachmentInvalidReason?: TranslationPaths | null;
+
+    /** Determines the title of the invalid reason modal */
+    attachmentInvalidReasonTitle?: TranslationPaths | null;
+
+    /** Ref to the submit button */
+    submitRef?: RefObject<View | HTMLElement>;
+
+    /** Determines if the delete receipt confirm modal is visible or not */
+    isDeleteReceiptConfirmModalVisible?: boolean;
+
+    /** Optional callback to fire when we want to preview an image and approve it for use. */
+    onConfirm?: (file: FileObject) => void;
+
+    /** Callback triggered when the modal is closed */
+    onClose?: (options?: OnCloseOptions) => void;
+
+    /** Callback triggered when the confirm modal is closed */
+    onConfirmModalClose?: () => void;
+
+    /** Callback triggered when the delete receipt modal is shown */
+    onRequestDeleteReceipt?: () => void;
+
+    /** Callback triggered when the delete receipt is confirmed */
+    onDeleteReceipt?: () => void;
+
+    /** Optional callback to fire when we want to do something after attachment carousel changes. */
+    onCarouselAttachmentChange?: (attachment: Attachment) => void;
+
+    /** Optional callback to fire when we want to validate the file. */
+    onValidateFile?: OnValidateFileCallback;
+};
+
+function AttachmentModalBaseContent({
+    source = '',
+    attachmentID,
+    fallbackSource,
+    file: fileProp,
+    originalFileName = '',
+    isAuthTokenRequired = false,
+    maybeIcon = false,
+    headerTitle: headerTitleProp,
+    type,
+    draftTransactionID,
+    iouAction,
+    iouType: iouTypeProp,
+    accountID,
+    attachmentLink = '',
+    allowDownload = false,
+    isTrackExpenseAction = false,
+    report,
+    reportID,
+    isReceiptAttachment = false,
+    isWorkspaceAvatar = false,
+    canEditReceipt = false,
+    canDeleteReceipt = false,
+    isLoading = false,
+    shouldShowNotFoundPage = false,
+    shouldDisableSendButton = false,
+    shouldDisplayHelpButton = true,
+    fallbackRoute,
+    isDeleteReceiptConfirmModalVisible = false,
+    isAttachmentInvalid = false,
+    attachmentInvalidReason,
+    attachmentInvalidReasonTitle,
+    submitRef,
+    onClose = () => Navigation.goBack(fallbackRoute),
+    onConfirm,
+    onConfirmModalClose,
+    onRequestDeleteReceipt,
+    onDeleteReceipt,
+    onCarouselAttachmentChange = () => {},
+    onValidateFile,
+}: AttachmentModalBaseContentProps) {
+    const styles = useThemeStyles();
+    const {windowWidth} = useWindowDimensions();
+    const {shouldUseNarrowLayout} = useResponsiveLayout();
+
+    const [sourceState, setSourceState] = useState<AvatarSource>(() => source);
+    const sourceForAttachmentView = sourceState || source;
+    useEffect(() => {
+        setSourceState(() => source);
+    }, [source]);
+
```

Comment:
I have no idea what this part is doing but it also feels a little weird

Reply:
Added a comment, which hopefully explains this correctly. Not totally sure if it's correct, as this is not my code